### PR TITLE
[Keyboard] Add Quarkeys Z67 PCB with solder and hotswap versions

### PR DESCRIPTION
**Pull request by tommyamoszhao**
<-- This is a 65% 67 keys PCB, designed and made by Qaurkeys Studio -->
<-- Solder version with different layout variants, through-hole for per-key LEDs -->
<-- Hotswap version with ansi 65% layout, per-key RGBs -->

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/quarkeys/z67/hotswap/rules.mk [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/quarkeys/z67/hotswap/rules.mk:13]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/quarkeys/z67/hotswap/rules.mk [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/quarkeys/z67/hotswap/rules.mk:12]:
```suggestion
BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no	    # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
```

File: keyboards/quarkeys/z67/solder/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+     * `--------+--------+--------+------------------------------------------+------+------+------+------+------+------'     
+     */
+	[1] = LAYOUT(
+		QK_BOOT, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,   KC_F12,   KC_BSPC, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_PGUP, KC_HOME, KC_UP,   KC_END,  KC_DEL,  KC_BSLS, KC_7,    KC_8,    KC_9,    RGB_TOG,  RGB_MOD,  RGB_VAI,           KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_PGDN, KC_LEFT, KC_DOWN, KC_RGHT, KC_PLUS, KC_MINS, KC_4,    KC_5,    KC_6,    RGB_SAD,  RGB_SAD,  RGB_VAD,  KC_TRNS,          KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_MPLY, KC_MUTE, KC_VOLD, KC_VOLU, KC_ASTR, KC_SLSH, KC_1,    KC_2,    KC_3,     KC_DOT,   KC_TRNS,           KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_0,                      KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS),
+
+	/* [2]
```
Comment (drashna) [inline keyboards/quarkeys/z67/solder/keymaps/default/keymap.c:58]: Empty layouts should be removed.  The same can be done with via keymaps too, as they're automatically filled with empty values, anymore.

File: keyboards/quarkeys/z67/solder/rules.mk [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+BOOTMAGIC_ENABLE = yes		# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes		# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes		# Audio control and System control(+450)
+CONSOLE_ENABLE = no			# Console for debug(+400)
+COMMAND_ENABLE = no  		# Commands for debug and configuration
+SLEEP_LED_ENABLE = no  		# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes			# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no 		# Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
+AUDIO_ENABLE = no
```
Comment (drashna) [inline keyboards/quarkeys/z67/solder/rules.mk:16]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes
AUDIO_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/quarkeys/z67/solder/keymaps/default/keymap.c (hunk 3 above)
Comment (tommyamoszhao) [reply keyboards/quarkeys/z67/solder/keymaps/default/keymap.c:58]: Thank you so much. I'll keep the layouts this way and so on for the future boards

**Review by drashna — APPROVED** [review]

File: keyboards/quarkeys/z67/solder/rules.mk [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic lite
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/solder/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic lite
```

File: keyboards/quarkeys/z67/solder/rules.mk [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
+AUDIO_ENABLE = no
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/solder/rules.mk:15]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/quarkeys/z67/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Quarkeys Z67 Keyboard
+
+A 65% Gasket keyboard made by Quarkeys Studio, third of our projects.
+
+* Keyboard Maintainer: [Quarkeys Studio](www.quarkeys.com)
+* Hardware Supported: Z67-YUN
+* Hardware Specs: Solder version with throught hole for LEDs
+                  Hotswap version with per-key RGBs, layer & Capslock indicators
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/readme.md:8]:
```suggestion
* Hardware Specs: Solder version with throught hole for LEDs, hotswap version with per-key RGBs, layer & Capslock indicators
```

File: keyboards/quarkeys/z67/hotswap/rules.mk [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes   # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no	    # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/hotswap/rules.mk:15]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/quarkeys/z67/hotswap/config.h [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+/ GNU General Public License for more details.
+/ You should have received a copy of the GNU General Public License
+/ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#include "config_common.h"
+
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/hotswap/config.h:19]:
```suggestion
```

File: keyboards/quarkeys/z67/hotswap/rules.mk [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/hotswap/rules.mk:6]:
```suggestion
```

File: keyboards/quarkeys/z67/solder/config.h [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+/ GNU General Public License for more details.
+/ You should have received a copy of the GNU General Public License
+/ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#include "config_common.h"
+
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 16
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/solder/config.h:19]:
```suggestion
```

File: keyboards/quarkeys/z67/hotswap/info.json [hunk 12]
```diff
@@ -0,0 +1,88 @@
+{
+    "keyboard_name": "Z67 Hotswap", 
+    "manufacturer": "Quarkeys Stuidio",
+    "url": "www.quarkeys.com", 
+    "maintainer": "TommyZ", 
+    "usb": {
+        "vid": "0x8490",
+        "pid": "0x4102",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/hotswap/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/quarkeys/z67/solder/info.json [hunk 13]
```diff
@@ -0,0 +1,92 @@
+{
+    "keyboard_name": "Z67 Solder", 
+    "manufacturer": "Quarkeys Stuidio",
+    "url": "www.quarkeys.com", 
+    "maintainer": "TommyZ", 
+    "usb": {
+        "vid": "0x8490",
+        "pid": "0x4101",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/solder/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/quarkeys/z67/solder/rules.mk [hunk 14]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/quarkeys/z67/solder/rules.mk:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by tommyamoszhao** [comment]
Hi, is there any request still pending? @fauxpark I think every request's been full filled. Let me know, appreciate. :)

**Review by fauxpark — APPROVED** [review]

**Pull request merged**